### ISSUE TITLE
Proposed change causes infinite loop

### ISSUE THREAD

**Issue by Stegallo** [sourcery-ai/sourcery#109]
### Issue description or question

when using the github app, sourcery proposes a change that changes the actual behavior of the code, resulting in the program to never terminate.

full code example:

```
class Sample:
    def __init__(self):
        self.__input = 0

    def __iteration_function(self):
        if self.__input < 10:
            self.__input += 1

    def run(self, function):
        while True:
            before = str(self.__input)
            function()
            if before == str(self.__input):
                break

            print(self.__input)

    def main(self):
        self.run(self.__iteration_function)


s = Sample()
s.main()
```

link to the actual generated PR:
[sample class (Sourcery refactored)](https://github.com/Stegallo/adventofcode/pull/122)

### Sourcery Version

github app

The proposed refactor in run() method moves `before = str(self.__input)` just before the while, causing the program to never exit.
My understanding is that sourcery detects the assignment to var before as not dependent by the instructions inside the loop. However, since function is actually operating on the instance variable self.__input, this assumption is no longer true.

**Comment by Hellebore** [comment]
Fixed in next release.

**Comment by Hellebore** [comment]
Fixed with release of 0.8.6
